### DISCUSSION TITLE
Regarding conditionally mange updates on server side through socket

### DISCUSSION THREAD

**Discussion by Jeelkumar-patel** [socketio/socket.io#4823] [Q&A]
Hey,
Thanks in Advance
I have one question. I want conditionally send update to all connected client.
Like Ex In my app there is many projects which have multiple user to assigned. so each project have multiple user. and for maintaining data I have used firebase firestore. Tiil now my whole data manage with Firebase Firestore snapshot but now I want to established socket in that for real time update. and also I want to send update to particular user who is assigned to that project. not other who is connected to socket. How to implement this type of flow with socket.

**Comment by darrachequesne** [comment]
Hi!

I think this is a good use case for rooms:

```js
io.on("connection", (socket) => {
  socket.join("some room");
});
```

And then later:

```js
io.to("some room").emit("some event");
```

Reference: https://socket.io/docs/v4/rooms/

**Reply by Jeelkumar-patel** [reply]
It's true, but I have a question regarding the rooms. If my project list grows in the future, managing many rooms could become a challenge, since each project has its own room.

**Reply by darrachequesne** [reply]
> managing many rooms could become a challenge

You mean, from a performance point of view? Or within the code itself?

In any case, if you have any idea how we could improve your dev experience, please suggest it here.
